telingo/theory/body.py

```diff
@@ -646,10 +738,104 @@ def create_atom(rep, add_formula, positive):
+            lhs = create_path(args[0],add_formula, False)
+            rhs = create_path(args[1],add_formula, False)
+            if rep.name == "+":
+                return add_formula(ChoicePath(lhs,rhs))
+            else:
+                assert(rep.name == ";;")
+                return add_formula(SequencePath(lhs,rhs))
+        elif rep.name in g_path_unary_operators:
+            if check:
+                raise RuntimeError("invalid dynamic formula: ".format(rep))
+            if rep.name == "?":
+                arg = create_path(args[0],add_formula, True)
+                return add_formula(CheckPath(arg))
+            else:
+                assert(rep.name == "*")
+                arg = create_path(args[0],add_formula, False)
+                return add_formula(KleeneStarPath(arg))
+        elif rep.name == "&":
+            arg = rep.arguments[0]
+            if arg.type == _clingo.TheoryTermType.Symbol:
+                if not check and arg.name == "true":
+                    return add_formula(SkipPath())
+                if check and arg.name == "true" or arg.name == "false":
+                    return add_formula(BooleanConstant(arg.name == "true"))
+                else:
+                    raise RuntimeError("unknown identifier: ".format(rep))
+            else:
+                raise RuntimeError("invalid dynamic formula: ".format(rep))
+        #this case is probably impossible 
+        else:
+            return create_atom(rep, add_formula, True)
+    else:
+        raise RuntimeError("invalid dynamic formula: ".format(rep))
+
+def create_dynamic_formula(rep, add_formula):
+    """
+    Returns the dynamic formula corresponding the given theory term.
+
+    Throws an error if rep it is not a valid formula.
+
+    Arguments:
+    rep         -- Theory term to translate.
+    add_formula -- Callback to add resulting formuals.
+    """
+    if rep.type == _clingo.TheoryTermType.Symbol:
+        return create_atom(rep, add_formula, True)
+    elif rep.type == _clingo.TheoryTermType.Function:
+        args = rep.arguments
+        if rep.name in g_del_operators:
+            lhs = create_path(args[0], add_formula, False)
+            rhs = create_dynamic_formula(args[1], add_formula)
+            if rep.name == ".>*":
+                return add_formula(BoxFormula(lhs,rhs))
+            else:
+                assert(rep.name == ".>?")
+                return add_formula(DiamondFormula(lhs,rhs))
+        elif rep.name == "&":
+            arg = rep.arguments[0]
+            if arg.type == _clingo.TheoryTermType.Symbol:
+                if arg.name == "true" or arg.name == "false":
+                    return add_formula(BooleanConstant(arg.name == "true"))
+                elif arg.name == "final":
+                    return add_formula(BoxFormula(add_formula(SkipPath()),add_formula(BooleanConstant(False))))
+                else:
+                    raise RuntimeError("unknown identifier: ".format(rep))
```

Comment:
There is no `{}` in the format string but an argument is passed in. Looks like this is repeated below too.